src/main/java/com/exasol/adapter/dynamodb/dynamodbmetadata/DynamodbTableMetadata.java

```diff
@@ -9,9 +9,9 @@
  */
 public class DynamodbTableMetadata implements Serializable {
     private static final long serialVersionUID = 7898591572006230212L;
-    private final DynamodbIndex primaryIndex;
-    private final List<DynamodbIndex> localIndexes;
-    private final List<DynamodbIndex> globalIndexes;
+    private final DynamodbPrimaryIndex primaryIndex;
```

Comment:
You can use interface here, I think. I like it somehow more as it was before changes

Reply:
But the primary index is always a `DynamodbPromaryIndex`. Using  the interface here would loose the type information.